api/defaults/defaults.go

```diff
@@ -34,6 +34,10 @@ const (
 	// DefaultIdleTimeout is a default idle connection timeout.
 	DefaultIdleTimeout = 30 * time.Second
 
+	// DefaultDialTimeout is the default time to wait for a connection to be
+	// established.
+	DefaultDialTimeout = 5 * time.Second
```

Comment:
I'd either move this out of `api/defaults` or name it something like `DeafultAppDialTimeout` since you're only ever using it when dialing apps.

Honestly having a `defaults` package like this was probably never a good idea, it becomes a dumping ground for completely unrelated configuration parameters.